### PR TITLE
Switches to codecov GH action

### PR DESCRIPTION
**Pull request by m1kola**
**Description of the change:**

Switches the repo to use codecov GH action.

**Motivation for the change:**

In #639 (after [Codecov bash uploader vulnerability](https://about.codecov.io/security-update/)) this repo was switched to use copy-pasted bash script.

Since then many things changed:

* [`codecov/codecov-action`](https://github.com/codecov/codecov-action) no longer uses bash uploader
* This action [validates sha sums](https://github.com/codecov/codecov-action/blob/eaaf4bedf32dbdc6b720b63067d99c4d77d6047d/src/validate.ts#L26-L62) 

I think it is safe to switch back to this GH action.

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage 
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/docs` 
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

File: .github/workflows/unit.yaml [hunk 1]
```diff
@@ -13,10 +13,13 @@ jobs:
   unit:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
```
Comment (m1kola) [inline .github/workflows/unit.yaml:16]: I wanted to update it with rest of the actions in #1117, but bash script fails with v3.

File: .github/workflows/unit.yaml [hunk 2]
```diff
@@ -13,10 +13,13 @@ jobs:
...
       - uses: actions/setup-go@v4
         with:
           go-version-file: "go.mod"
       - run: make unit
       - run: sed -i'' "s:^github.com/$GITHUB_REPOSITORY/::" coverage.out
-      - run: .github/workflows/codecov.sh -Z -f coverage.out
+      - uses: codecov/codecov-action@v3
+        with:
+          files: coverage.out
+          fail_ci_if_error: true
```
Comment (m1kola) [inline .github/workflows/unit.yaml:25]: Equivalent of `-Z` from the bash script.

File: .github/workflows/unit.yaml [hunk 3]
```diff
@@ -13,10 +13,14 @@ jobs:
...
+          functionalities: fixes
```
[the remark this answers is not in the document]
Comment (m1kola) [reply .github/workflows/unit.yaml:26]: Alternatively we can remove this line and estabilish a new baseline in master without enabling these fixes.

[the remark this answers is not in the document]
Comment (grokspawn) [reply .github/workflows/unit.yaml:26]: Honestly, good with either approach.  Using 'fixes' right now seems generally good?

**Review by grokspawn — APPROVED** [review]

File: .github/workflows/unit.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (m1kola) [reply .github/workflows/unit.yaml:26]: Yeah, I think `fixes` gives exactly the same result as it was with the previous uploader. After adding `fixes` the diff between `master` coverage and this PR is empty (see results [here](https://app.codecov.io/github/operator-framework/operator-registry/commit/5c43e7b0ed0386a494861b9bb9b15b259858fa7b)).

[the remark this answers is not in the document]
Comment (m1kola) [reply .github/workflows/unit.yaml:26]: > After adding `fixes` the diff between `master` coverage and this PR is empty (see results [here](https://app.codecov.io/github/operator-framework/operator-registry/commit/5c43e7b0ed0386a494861b9bb9b15b259858fa7b)).

Ah, interesting. Yesterday after running the tests the diff was empty and now it says +1.16%. Probably codecov was still processing when I opened the link.

Anyway - I think we should keep `fixes` for now.

**Pull request merged**